### PR TITLE
Add react-router, set up App.js

### LINKED ISSUE #408: Add react-router
We need react-router so we can re-use the front-end code for profiles from xPRO. It's not in bootcamps already because bootcamps predates our addition of react-router to the project cookie-cutter.

┆Attachments: <a href="https://github.com/mitodl/bootcamp-ecommerce/pull/412">bootcamp-ecommerce #412 Add react-router, set up App.js</a>

### PR DESCRIPTION
**Pull request by noisecapella**
#### What are the relevant tickets?
Fixes #408 

#### What's this PR do?
Adds App.js and react-router so we can add future react pages similar to our other projects

#### How should this be manually tested?
Nothing should look different. In particular you should be able to go to `/pay` and everything should look the same

### CODE REVIEW

File: static/js/entry/payment.js [hunk 1]
```diff
@@ -1,20 +1,29 @@
 // @flow
+import { createBrowserHistory } from "history"
```
Comment (mbertrand) [inline static/js/entry/payment.js:2]: What do you think of changing this filename to `root.js` and updating the webpack config to reflect that?

Comment (noisecapella) [reply static/js/entry/payment.js:2]: :+1:

**Comment by annagav** [comment]
Everything seems to work well for me. This is probably not related to this PR but when I try to logout it takes me to the admin logout page. Should I open a separate issue for that?

**Comment by noisecapella** [comment]
@annagav Yes, make a separate issue for that. Anything else for this PR?

**Pull request merged**